app/routes/exercises.rb

```diff
@@ -66,6 +66,28 @@ class Exercises < Core
         redirect '/dashboard'
       end
 
+      post '/exercises/:key/request_help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may request help."
+          redirect '/'
+        end
+        exercise.request_help!(current_user)
+        flash[:success] = "You have requested for help on this submission"
+        redirect "/submissions/%s" % exercise.submissions.last.key
+      end
+
+      post '/exercises/:key/remove_help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may remove help."
```

Comment:
The phrase "remove help" is a bit confusing. I think we need to think through the terminology of how people will be referring to this.